head.go

```diff
@@ -329,6 +323,18 @@ func (h *Head) loadWAL(r *wal.Reader) error {
 	)
 	wg.Add(n)
 
+	defer func() {
+		// For CorruptionErr ensure to terminate all workers before exiting.
+		if _, ok := err.(*wal.CorruptionErr); ok {
```

Comment:
Should we check for `err != nil` before?

Reply:
I guess it changed again the in last commit.

Reply:
don't think it is necessary, but doesn't hurt so I updated it.

Reply:
confirmed that typecasting on nil interface doesn't panic
https://play.golang.org/p/plfVNzBEclS